wasi/wasi.go

```diff
@@ -277,16 +323,28 @@ func (w *WASIEnvironment) fd_close(ctx *wasm.HostFunctionCallContext, fd uint32)
 	return ESUCCESS
 }
 
-func args_sizes_get(ctx *wasm.HostFunctionCallContext, argcPtr uint32, argvPtr uint32) (err Errno) {
-	// not implemented yet
-	binary.LittleEndian.PutUint32(ctx.Memory.Buffer[argcPtr:], 0)
-	binary.LittleEndian.PutUint32(ctx.Memory.Buffer[argvPtr:], 0)
-	return 0
+func (w *WASIEnvironment) args_sizes_get(ctx *wasm.HostFunctionCallContext, argsCountPtr uint32, argsBufSizePtr uint32) Errno {
+	if !ctx.Memory.PutUint32(argsCountPtr, uint32(len(w.args.strings))) {
+		return EINVAL
+	}
+	if !ctx.Memory.PutUint32(argsBufSizePtr, w.args.totalBufSize) {
+		return EINVAL
+	}
+
+	return ESUCCESS
 }
 
-func args_get(*wasm.HostFunctionCallContext, uint32, uint32) (err Errno) {
-	// not implemented yet
-	return
+func (w *WASIEnvironment) args_get(ctx *wasm.HostFunctionCallContext, argsPtr uint32, argsBufPtr uint32) (err Errno) {
```

Comment:
same comment about.. godoc (`// args_get does blah... // See https://link`) particularly where there are pointers involved, it is helpful to have a bullet per arg saying what they are supposed to represent and what we are doing with them as a result.

Ex. as there's no link I will guess :D

seems like argsPtr is where to write memory locations of each arg (as uint64). I'm not sure how this works without writing the count first (ex how does the other side know when to stop reading?), but maybe there's something in docs we can say as to how that works. 

seems like argsBufPtr is where we write the null terminated strings. 

Seems that the locations aren't assumed to be sequential in the buffer, just we do that for convenience.

Reply:
easiest way is to adapt in the text from here and link it as it also explains why the count isn't returned. notably, we should avoid using words like "array" though as we can narrow more specifically or less confusingly (ex the var name is but, so in s/array/buffer/ at least)

```
  ;;; Read command-line argument data.
  ;;; The size of the array should match that returned by `args_sizes_get`
```

https://github.com/WebAssembly/WASI/blob/snapshot-01/phases/snapshot/witx/wasi_snapshot_preview1.witx#L15-L29

Reply:
Added docs. PTAL